### PR TITLE
#10480 - update sormas setup and development environment documentation

### LINKED ISSUE #10480: Improve dev setup documentation [1]
### Problem Description

We often got diffuse problems reported that setting up the dev environment. @valentinmikleuvg documented his setup process in a docx file.

### Proposed Solution

- Check the provided document what is different to existing documentation (unfortunately no diff to current documentation given).
- Improve [dev environment setup documentation](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/DEVELOPMENT_ENVIRONMENT.md).

### Additional Information

[SORMAS DEV SETUP .docx](https://github.com/hzi-braunschweig/SORMAS-Project/files/9673165/SORMAS.DEV.SETUP.docx)

Also have a look into https://github.com/hzi-braunschweig/SORMAS-Project/pull/2375 to discuss with @MartinWahnschaffe how to proceed there.

### PR DESCRIPTION
**Pull request by BarnaBartha**
Fixes #10480

### CODE REVIEW

File: docs/DEVELOPMENT_ENVIRONMENT.md [hunk 1]
```diff
@@ -5,27 +5,33 @@ This step-by-step guide explains how to set up your development environment, usi
...
 
-## Step 4: Install and Configure Your IDE
+## Step 3: Install Maven & Ant
+Download and install Maven for your operating system, see [binaries](https://dlcdn.apache.org/maven/maven-3/3.6.3/binaries/)
+*IMPORTANT*: M2_HOME environment variable needs to be set. By default, for newer version, it is set to MAVEN_HOME. But Ant script is looking for M2_HOME, please refer to the [official documentation](https://maven.apache.org/install.html)
+
+Download and install Ant, it can be done from [Ant site](https://ant.apache.org/bindownload.cgi) or with packages from your Linux distribution.
+
+## Step 4: Install a Local SORMAS Server
+Please follow the [Server Installation Instructions](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/SERVER_SETUP.md#sormas-installation) to set up a local SORMAS instance that you will use to test your code. Alternatively, you can also use [Maven Cargo](../sormas-cargoserver/README.md), or a [Docker installation](SERVER_DOCKER_SETUP.md) (not recommended at this time).
```
Comment (StefanKock) [inline docs/DEVELOPMENT_ENVIRONMENT.md:26]: Change to relative path `[Server Installation Instructions](docs/SERVER_SETUP.md#sormas-installation)`

File: docs/DEVELOPMENT_ENVIRONMENT.md [hunk 2]
```diff
@@ -127,6 +134,8 @@ Optional, but strongly recommended:
 
 7. For eclipse formatted plugin, there is an issue for Idea: <https://plugins.jetbrains.com/plugin/6546-eclipse-code-formatter> - `cannot save settings Path to custom eclipse folder is not valid` - it works only when settings were saved from down to up. And not vice versa.
 
+If something is still not working, try to clean up (delete all from domains/sormas/autodeploy, domains/sormas/applications, domains/sormas/generated, and domains/sormas/osgi-cache) try to build again by executing `mvn clean install -DskipTests` on the `sormas-base` module
```
Comment (StefanKock) [inline docs/DEVELOPMENT_ENVIRONMENT.md:137]: Stop the payara domain before cleaning, run Ant `deploy-serverlibs` to update libs before starting the domain and deploying again

File: docs/SERVER_SETUP.md [hunk 3]
```diff
@@ -65,7 +65,7 @@ You can check your Java version from the shell/command line using: ``java -versi
 * Install the "temporal tables" extension for Postgres (<https://github.com/arkhipov/temporal_tables>)
   * **Windows**: Download the latest version for your Postgres version: <https://github.com/arkhipov/temporal_tables/releases/latest>, then copy the DLL from the project into the PostgreSQL's lib directory and the .sql and .control files into the directory share\extension.
   * **Linux** (see <https://github.com/arkhipov/temporal_tables#installation)>
-
+* Add the PostgreSQL path (/etc/PostgreSQL/10/bin) to Environment Variables
```
Comment (StefanKock) [inline docs/SERVER_SETUP.md:68]: Why is this line added here? The script belongs to the temporal tables topic

File: README.md [hunk 4]
```diff
@@ -64,7 +64,7 @@ If you want to learn more about the development and contribution process, settin
 
 If you want to set up a SORMAS instance for production, testing or development purposes, please refer to the following guides:
 * [Installing a SORMAS Server](docs/SERVER_SETUP.md)
-* [Installing a SORMAS Server for Development](docs/SERVER_DEV_SETUP.md)
+* [Installing a SORMAS Server for Development](docs/SERVER_DOCKER_SETUP.md)
```
Comment (StefanKock) [inline README.md:67]: Better like this:
```
* [Installing a SORMAS Server](docs/SERVER_SETUP.md)
* [Updating a SORMAS Server](docs/SERVER_UPDATE.md)
* [Installing Development environment](docs/DEVELOPMENT_ENVIRONMENT.md)
* [Creating a Demo Android App](docs/DEMO_APP.md)
```

**Review by StefanKock — CHANGES_REQUESTED** [review]
The changes look good in general, only small findings

File: docs/SERVER_SETUP.md (hunk 3 above)
Comment (BarnaBartha) [reply docs/SERVER_SETUP.md]: where should it be added?

Comment (StefanKock) [reply docs/SERVER_SETUP.md]: Just after the script path I would say

Comment (BarnaBartha) [reply docs/SERVER_SETUP.md]: the script path - what is that ?

Comment (StefanKock) [reply docs/SERVER_SETUP.md]: The script block (line 69ff) I meant but did not type.
